plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestTransactionLogAccess.java

```diff
@@ -736,6 +738,45 @@ public void testTableSnapshotsCacheDisabled()
                         "00000000000000000014.json", 2));
     }
 
+    @Test
+    public void testTableSnapshotsActiveDataFilesCacheExpiration()
+            throws Exception
+    {
+        String tableName = "person";
+        Path tableDir = new Path(getClass().getClassLoader().getResource("databricks/" + tableName).toURI());
+        DeltaLakeConfig shortLivedActiveDataFilesCacheConfig = new DeltaLakeConfig();
+        shortLivedActiveDataFilesCacheConfig.setDataFileCacheTtl(new Duration(10, TimeUnit.MILLISECONDS));
+        setupTransactionLogAccess(tableName, tableDir, shortLivedActiveDataFilesCacheConfig);
+
+        List<AddFileEntry> addFileEntries = transactionLogAccess.getActiveFiles(tableSnapshot, SESSION);
+        assertEquals(addFileEntries.size(), 12);
+        assertEquals(
+                hdfsEnvironment.getOpenCount(),
```

Comment:
Is it really "open" count? The current implementation seems counting "get" file system. 

As you may already know, there's already `AccessTrackingFileSystem`. Duplicated `TrackingHdfsEnvironment` (having the same method with different purpose) looks little confusing to me.

Reply:
> Is it really "open" count? The current implementation seems counting "get" file system.

Indeed. I modified the getter to be named `getAccessedPathNames()`.

The new utility class `TrackingHdfsEnvironment` is used for the special use-cases where the Trino code makes method calls like:

```
hdfsEnvironment.getFileSystem(...)
```

where we previously didn't control the returned output.

Probably it would be better to wrap also the returned `FileSystem` instance with `AccessTrackingFileSystem` and do the "open" counting on a statically available map.